tools/image_processing/graphicsmagick/montage.xml

```diff
@@ -1,41 +1,49 @@
-<tool id="graphicsmagick_image_montage" name="Image Montage" version="@VERSION@">
+<tool id="graphicsmagick_image_montage" name="Image Montage" version="@VERSION@+galaxy1">
     <description></description>
     <macros>
         <import>macros.xml</import>
     </macros>
     <expand macro="requirements" />
     <command detect_errors="aggressive">
 <![CDATA[
-        ln -s '${input}' input_existing.jpg
-        &&
-        ln -s '${input_new}' input_new.jpg
-        &&
-        montage input_existing.jpg input_new.jpg -geometry +0+0 -background none -tile 1x2 temp.jpg
+#for idx, data in enumerate($input):
+ln -s '${data}' ${idx}.${data.ext};
+#end for
+
+montage
+
+#for idx, data in enumerate($input):
+    ${idx}.${data.ext}
+#end for
+    -geometry +0+0
+    -background none
+    -tile ${width}x
+    temp.png
 ]]>
     </command>
 
     <inputs>
-        <param format="jpg" name="input" type="data" label="Existing Image" />
-        <param format="jpg" name="input_new" type="data" label="New Image" />
+        <param format="jpg,png,bmp,gif,svg,eps,tiff" name="input" type="data" multiple="true" label="Images" />
```

Comment:
I like that, but should this be an option? X inputs --> 1 output or X inputs --> X outputs?

Reply:
I had not considered that use case. Do you think it's common, or maybe should the user use something like a collection splitting operation ahead of this tool?

Reply:
I guess this can work!